src/relentless/model/potential/dihedral.py

```diff
@@ -0,0 +1,482 @@
+    +-------------+--------------------------------------------------+
+    | ``k_2``     | Second fitting coefficient :math:`k_2`.          |
+    +-------------+--------------------------------------------------+
+    | ``k_3``     | Third fitting coefficient :math:`k_3`.           |
+    +-------------+--------------------------------------------------+
+    | ``k_4``     | Fourth fitting coefficient :math:`k_4`.          |
+    +-------------+--------------------------------------------------+
+
+    Parameters
+    ----------
+    types : tuple[str]
+        Types.
+    name : str
+        Unique name of the potential. Defaults to ``__u[id]``, where ``id`` is the
+        unique integer ID of the potential.
+
+    Attributes
+    ----------
+    coeff : :class:`DihedralParameters`
+        Parameters of the potential for each type.
+
+    Examples
+    --------
+    OPLS Dihedral::
+
+        >>> u = relentless.potential.dihedral.OPLSDihedral(("A",))
+        >>> u.coeff["A"].update({'k1': 1.740, 'k2': -0.157, 'k3': 0.279, 'k4': 0.00})
+
+    """
+
+    def __init__(self, types, name=None):
+        super().__init__(keys=types, params=("k1", "k2", "k3", "k4"), name=name)
+
+    def energy(self, type_, phi):
+        """Evaluate dihedral energy."""
+        params = self.coeff.evaluate(type_)
+        k1 = params["k1"]
+        k2 = params["k2"]
+        k3 = params["k3"]
+        k4 = params["k4"]
+
+        phi, u, s = self._zeros(phi)
+
+        u = 0.5 * (
+            k1 * (1 + numpy.cos(phi))
+            + k2 * (1 - numpy.cos(2 * phi))
+            + k3 * (1 + numpy.cos(3 * phi))
+            + k4 * (1 - numpy.cos(4 * phi))
+        )
+
+        if s:
+            u = u.item()
+        return u
+
+    def force(self, type_, phi):
+        """Evaluate dihedral force."""
+        params = self.coeff.evaluate(type_)
+        k1 = params["k1"]
+        k2 = params["k2"]
+        k3 = params["k3"]
+        k4 = params["k4"]
+
+        phi, f, s = self._zeros(phi)
+
+        f = -0.5 * (
+            -k1 * numpy.sin(phi)
+            + 2 * k2 * numpy.sin(2 * phi)
+            - 3 * k3 * numpy.sin(3 * phi)
+            + 4 * k4 * numpy.sin(4 * phi)
+        )
+
+        if s:
+            f = f.item()
+        return f
+
+    def _derivative(self, param, phi, k1, k2, k3, k4):
+        r"""Evaluate dihedral derivative with respect to a variable."""
+        phi, d, s = self._zeros(phi)
+
+        if param == "k1":
+            d = 0.5 * (1 + numpy.cos(phi))
+        elif param == "k2":
+            d = 0.5 * (-1 - numpy.cos(2 * phi))
+        elif param == "k3":
+            d = 0.5 * (1 + numpy.cos(3 * phi))
+        elif param == "k4":
+            d = 0.5 * (-1 - numpy.cos(4 * phi))
+        else:
+            raise ValueError("Unknown parameter")
+
+        if s:
+            d = d.item()
+        return d
+
+
+class RyckaertBellemansDihedral(DihedralPotential):
+    r"""Cosine squared dihedral potential.
```

Comment:
Fix up the short class description, maybe this?
```suggestion
    r"""Ryckaert–Bellemans dihedral potential.
```